source/app.d

```diff
@@ -89,19 +91,20 @@ class ImageFitness
     }
 }
 
-
-void draw()
+void draw(string input,float mutation = 0.99)
 {
-    ImageFitness fitness = new ImageFitness("mona4.jpg");
-    const uint NUMBER_OF_CIRCLES = 10;
-    geneticAlgorithm!(fitness)(fitness.populationSize * NUMBER_OF_CIRCLES, 0.0, 50, 0.99);
+	ImageFitness fitness = new ImageFitness(input);
+	const uint NUMBER_OF_RECTANGLES = 20;
+	geneticAlgorithm!(fitness)(fitness.populationSize * NUMBER_OF_RECTANGLES, 0.0, 50, mutation);
 }
 
 void main(string[] argv)
 {
-    if (argv.length > 1)
-    {
-        getoptions(argv);
-    }
-    draw();
+	try
+	{
+			auto data = getOptions(argv);
+			draw(data.input, data.probability);
```

Comment:
Obecně máš pravdu, ale jenom obalení callu do try není "používání výjimek". Náročnost výjimek vychází z nustnosti unwindovat stack a najít handler, což se ovšem dělá až při vyhození. Ovšem je pravda, že overhead overhead trochu přidává i vstoupení do try bloku, což se ale děje pouze v téhle části kódu.

Takže pokud neházíš výjimky třeba v každé iteraci nějaké vnitřní smyčky, tak je to v pohodě a nemáš všechno obalený do try bloku, tak je to v pohodě.

Každopádně já si myslím, že tenhle try blok je useless, protože, jak už jsem psal výše, si myslím, že špatný argumenty programu nejsou výjimečná situace.